src/pseudopeople/interface.py

```diff
@@ -29,7 +28,7 @@ def _generate_dataset(
     dataset_schema: DatasetSchema,
     source: Optional[Union[Path, str]],
     seed: int,
-    config: Optional[Union[Path, str, NestedDict]],
+    config: Optional[Union[Path, str, dict[str, Any]]],
```

Comment:
Can you create a ticket (if it doesn't already exist) to replace all instances of `Union` and `Optional` with `|`?